src/main/java/org/mockito/internal/framework/DefaultMockitoSession.java

```diff
@@ -43,7 +49,7 @@ public Throwable getFailure() {
                 return null;
             }
             public Object getTestClassInstance() {
-                return testClassInstance;
+                return testClassInstances.get(0);
```

Comment:
This is unfortunate. We probably have to fix #1232 first to improve our internal API and then base this PR on that change.